src/main/java/tutorease/address/logic/parser/EditContactCommandParser.java

```diff
@@ -14,33 +15,40 @@
 import java.util.Set;
 
 import tutorease.address.commons.core.index.Index;
-import tutorease.address.logic.commands.EditCommand;
-import tutorease.address.logic.commands.EditCommand.EditPersonDescriptor;
+import tutorease.address.logic.commands.EditContactCommand;
+import tutorease.address.logic.commands.EditContactCommand.EditPersonDescriptor;
 import tutorease.address.logic.parser.exceptions.ParseException;
 import tutorease.address.model.tag.Tag;
 
 /**
  * Parses input arguments and creates a new EditCommand object
  */
-public class EditCommandParser implements Parser<EditCommand> {
+public class EditContactCommandParser implements Parser<EditContactCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the EditCommand
      * and returns an EditCommand object for execution.
      *
      * @throws ParseException if the user input does not conform the expected format
      */
-    public EditCommand parse(String args) throws ParseException {
+    public EditContactCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ROLE,
+                        PREFIX_ADDRESS, PREFIX_TAG);
```

Comment:
Since we do not allow the change of roles, is there still a need to keep it as a prefix in the command?

Reply:
It was an addon to be honest. This will still allow user to include the prefix in the command but an error will prompt that user is not allowed to change the roles.

Reply:
Makes sense. If not users might try editing their role and not see that nothing will be changed. Good catch, we should add this to our UG too. LGTM!